### PR TITLE
Add `color_channels` option in in plot_probe_map

### LINKED ISSUE #2739: channel_ids in plot_probe_map function does not work
Hi,

The channel_ids argument in plot_probe_map (src/spikeinterface/widgets/probe_map.py) does not filter selected channels:

```
import spikeinterface as si  # import core only
import spikeinterface.extractors as se

local_path = si.download_dataset(remote_path="mearec/mearec_test_10s.h5")
recording, sorting_true = se.read_mearec(local_path)

fig, ax = plt.subplots(figsize=(10, 10))
si.plot_probe_map(recording, ax=ax, channel_ids=["1","2"], with_channel_ids=True, contacts_colors="r")
```

### PR DESCRIPTION
**Pull request by RobertoDF**
Fix #2739

### CODE REVIEW

**Comment by alejoe91** [comment]
Thanks @RobertoDF, I think it's a good solution. Can you add a code example here and a screenshot to show how it looks?

**Comment by RobertoDF** [comment]
Here it is:

```
# two probe groups in rec
bad_channel_ids, channel_labels = detect_bad_channels(high_passed_rec, dead_channel_threshold=-1)

color_channels = [channel_label_color_dict[label] for label in channel_labels]

si.plot_probe_map(raw_rec, color_channels=color_channels, with_channel_ids=False)
```
Screenshot 2024-04-22 at 13 24 09

File: src/spikeinterface/widgets/probe_map.py [hunk 1]
```diff
@@ -15,16 +15,16 @@ class ProbeMapWidget(BaseWidget):
     ----------
     recording: RecordingExtractor
         The recording extractor object
-    channel_ids: list
-        The channel ids to display
+    color_channels: list
```
Comment (alejoe91) [inline src/spikeinterface/widgets/probe_map.py:18]:
```suggestion
    color_channels: list or matplotlib-accepted color
```

File: src/spikeinterface/widgets/probe_map.py [hunk 2]
```diff
@@ -70,8 +70,15 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
             n = probe.get_contact_count()
             if dp.with_channel_ids:
                 text_on_contact = dp.recording.channel_ids[pos : pos + n]
+            if dp.color_channels is not None:
+                if len(dp.color_channels) > 1:
```
Comment (alejoe91) [inline src/spikeinterface/widgets/probe_map.py:74]:
```suggestion
                if isinstance(dp.color_channels, (list, np.ndarray)) and len(dp.color_channels) > 1:
```

File: src/spikeinterface/widgets/probe_map.py [hunk 3]
```diff
@@ -70,8 +70,15 @@ def plot_matplotlib(self, data_plot, **backend_kwargs):
...
+                    color = dp.color_channels[pos : pos + n]
+                elif len(dp.color_channels) == 1:
```
Comment (alejoe91) [inline src/spikeinterface/widgets/probe_map.py:76]:
```suggestion
                else:
```

**Review by alejoe91 — COMMENTED** [review]
@RobertoDF one last suggestion. I think that instead of passing a list of one elements, one easier option is to allow this:
```
plot_probe_map(rec, color_channels="r")
```

See implementation details in review :)

**Comment by RobertoDF** [comment]
definitely better 👍👍

**Comment by alejoe91** [comment]
@RobertoDF can you fix the conflict? Looks good to merge after that!

**Comment by RobertoDF** [comment]
ah yes, my mistake. hope it is ok now.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**